mirage-nat/unikernel.ml

```diff
@@ -46,16 +53,25 @@ module Main (R : Mirage_random.S) (M : Mirage_clock.MCLOCK) (P : Mirage_clock.PC
                         (match e with `Local -> "local" | `Gateway -> "gateway"));
             Lwt.return_unit
           | Ok dst ->
+            let more = ref [] in
             E.write eth dst `IPv4 (fun b ->
                 match Nat_packet.into_cstruct packet b with
-                | Ok n -> n
+                | Ok (n, adds) -> more := adds ; n
                 | Error e ->
                   Log.err (fun m -> m "error %a while Nat_packet.into_cstruct"
                               Nat_packet.pp_error e);
-                  0) >|= function
```

Comment:
will `0` trigger the `Error` case below? (if not, should we record an error condition? if yes, should we leave a comment for the next reviewer that this is how it works?)

Reply:
the contract^W signature of [`E.write` is `Cstruct.t -> int` for the `fill` function](https://github.com/mirage/mirage-protocols/blob/v4.0.1/src/mirage_protocols.mli#L32-L37), i.e. there's no way to error when calling `write` in the fill function. The return value (`int`) is used by the ethernet layer to determine how many bytes to send, if this case here is visited, an empty ethernet frame (i.e. only the header) will be transmitted (not actually sure whether it will be transmitted). this no-error in fill may actually be a design mistake we should re-think.

Reply:
Yeah, it seems like something that could fail.